src/disks.py

```diff
@@ -0,0 +1,98 @@
+from logging import getLogger
+
+import dbus
+from _dbus_bindings import PROPERTIES_IFACE
+
+logger = getLogger("yunohost.storage")
+
+
+UDISK_DRIVE_PATH = "/org/freedesktop/UDisks2/drives/"
+UDISK_DRIVE_IFC = "org.freedesktop.UDisks2.Drive"
+
+
+def _humaize(byte_size):
+    suffixes = "kMGTPEZYRQ"
+
+    byte_size = float(byte_size)
+
+    if byte_size < 1024:
+        return f"{byte_size}B"
+
+    for i, s in enumerate(suffixes, start=2):
+        unit = 1024**i
+        if byte_size <= unit:
+            return f"{(1024 * (byte_size / unit)):.1f} {s}B"
+
+
+def _query_udisks():
+    bus = dbus.SystemBus()
+    manager = dbus.Interface(
+        bus.get_object("org.freedesktop.UDisks2", "/org/freedesktop/UDisks2"),
+        "org.freedesktop.DBus.ObjectManager",
+    )
+
+    for name, dev in manager.GetManagedObjects().items():
+        if name.startswith(UDISK_DRIVE_PATH):
+            yield name, dev
+
+
+def _disk_infos(name: str, drive: dict, human_readable=False):
+    result = {
+        "name": name,
+        "model": drive["Model"],
+        "serial": drive["Serial"],
+        "removable": bool(drive["MediaRemovable"]),
+        "size": _humaize(drive["Size"]) if human_readable else drive["Size"],
+    }
+
+    if connection_bus := drive["ConnectionBus"]:
+        result["connection_bus"] = connection_bus
```

Comment:
Should those keys be translated when using the `human_readable` option?

Reply:
I think it's not necessary.